src/include/storage/storage_structure/lists/lists_update_store.h

```diff
@@ -76,49 +104,83 @@ class ListsUpdateStore {
     uint64_t getNumInsertedRelsForNodeOffset(
         ListFileID& listFileID, node_offset_t nodeOffset) const;
 
-    void readValues(
-        ListFileID& listFileID, ListHandle& listHandle, shared_ptr<ValueVector> valueVector) const;
+    void readValues(ListFileID& listFileID, ListHandle& listSyncState,
+        shared_ptr<ValueVector> valueVector) const;
+
+    bool hasAnyDeletedRelsInPersistentStore(ListFileID& listFileID, node_offset_t nodeOffset) const;
+
+    void updateRelIfNecessary(const shared_ptr<ValueVector>& srcNodeIDVector,
```

Comment:
Let's add a comment about the IfNecessary part here too. Something like: `This function is called ifNecessary because if the updated rel is stored in a column, e.g., if this rel tables is ONE-ONE, we don't need to update this rel in any lists.`

Reply:
Done.